### ISSUE TITLE
Placeholder for old PR #316

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#316]
Originally raised by @nojima in https://github.com/cuelang/cue/pull/316

**Comment by cueckoo** [comment]
_Original reply by @y-taka-23 in https://github.com/cuelang/cue/pull/316#issuecomment-603789035_

Thanks for the nice catch, but I'm afraid that, the first `-l` option will be overwritten by the second thus it results in a conflict 

```
prometheus: apiVersion: "v1" (from Service) & "apps/v1" (from Deployment)
```

Surely the following (space-delimited) is what you want.

```
$ cue import ./... -p kube -l '"\(strings.ToCamel(kind))" "\(metadata.name)"' -f
```

**Comment by cueckoo** [comment]
_Original reply by @nojima in https://github.com/cuelang/cue/pull/316#issuecomment-605393021_

Umm.. Test failed...

```
2020/03/28 04:21:24 $ cue cmd ls ./frontend/maitred
cmd must be run as one of its subcommands: unknown subcommand "ls"
Ensure commands are defined in a "_tool.cue" file.
Run 'cue help cmd' for known subcommands.
FAIL	cuelang.org/go/doc/tutorial/kubernetes	9.043s
```

**Comment by cueckoo** [comment]
_Original reply by @nojima in https://github.com/cuelang/cue/pull/316#issuecomment-605395160_

@y-taka-23
Thank you for your comment.

When I attempted the command you suggest, I got this error.

```
$ cue import ./... -p kube -l '"\(strings.ToCamel(kind))" "\(metadata.name)"' -f
labels must be expressions (-l foo -l 'strings.ToLower(bar)') or full paths (-l '"foo": "\(strings.ToLower(bar))":) : parser error in path "\"\\(strings.ToCamel(kind))\" \"\\(metadata.name)\"": expected label or ':', found 'EOF'
```

When I ran CUE with `-l` twice, I got the following file. The paths appeared to be concatenated without being overwritten. (I'm new to CUE, so I'm not sure I have a proper understanding of what `-l` means.)

```
package kube

service: prometheus: {
        apiVersion: "v1"
        kind:       "Service"
        ...
}
deployment: prometheus: {
        apiVersion: "apps/v1"
        kind:       "Deployment"
        metadata: name: "prometheus"
        ...
}
```

**Comment by cueckoo** [comment]
_Original reply by @y-taka-23 in https://github.com/cuelang/cue/pull/316#issuecomment-605433207_

@nojima Here is my minimum repro. Could you taste it?

```
$ cue version
cue version 0.0.15 darwin/amd64

$ cat sample.yaml
foo: foo-a
bar: bar-a
---
foo: foo-b
bar: bar-b

$ cue import ./... -p sample -l '"\(foo)"' -f
$ cat sample.cue
package sample

"foo-a": {
	foo: "foo-a"
	bar: "bar-a"
}
"foo-b": {
	foo: "foo-b"
	bar: "bar-b"
}

$ cue import ./... -p sample -l '"\(foo)"' -l '"\(bar)"' -f
$ cat sample.cue
package sample

"bar-a": {
	foo: "foo-a"
	bar: "bar-a"
}
"bar-b": {
	foo: "foo-b"
	bar: "bar-b"
}

$ cue import ./... -p sample -l '"\(foo)" "\(bar)"' -f
$ cat sample.cue
package sample

"foo-a": "bar-a": {
	foo: "foo-a"
	bar: "bar-a"
}
"foo-b": "bar-b": {
	foo: "foo-b"
	bar: "bar-b"
}
```

**Comment by cueckoo** [comment]
_Original reply by @nojima in https://github.com/cuelang/cue/pull/316#issuecomment-605456659_

@y-taka-23 
In my local environment, the result was different from yours...

```
$ cue version
cue version (devel) linux/amd64

$ git rev-parse HEAD
6830ce502cbcd25d81af5bdaa4af14d80f61f703

$ cat sample.yaml
foo: foo-a
bar: bar-a
---
foo: foo-b
bar: bar-b

$ cue import ./... -p sample -l '"\(foo)"' -f
$ cat sample.cue
package sample

"foo-a": {
        foo: "foo-a"
        bar: "bar-a"
}
"foo-b": {
        foo: "foo-b"
        bar: "bar-b"
}

$ cue import ./... -p sample -l '"\(foo)"' -l '"\(bar)"' -f
$ cat sample.cue
package sample

"foo-a": "bar-a": {
        foo: "foo-a"
        bar: "bar-a"
}
"foo-b": "bar-b": {
        foo: "foo-b"
        bar: "bar-b"
}


$ cue import ./... -p sample -l '"\(foo)" "\(bar)"' -f
labels must be expressions (-l foo -l 'strings.ToLower(bar)') or full paths (-l '"foo": "\(strings.ToLower(bar))":) : parser error in path "\"\\(foo)\" \"\\(bar)\"": expected label or ':', found 'EOF'
```

**Comment by cueckoo** [comment]
_Original reply by @nojima in https://github.com/cuelang/cue/pull/316#issuecomment-605456953_

↑ My repo was forked from 52cc7f11efb7567a2e9da3690a61fbaacbcca3bb.

**Comment by cueckoo** [comment]
_Original reply by @y-taka-23 in https://github.com/cuelang/cue/pull/316#issuecomment-605506118_

@nojima Aha, it 's just because my executable is pretty old. Sorry for confusing you.

**Comment by cueckoo** [comment]
_Original reply by @nojima in https://github.com/cuelang/cue/pull/316#issuecomment-605554653_

Oh, I understood there was an incompatible change...

**Comment by cueckoo** [comment]
_Original reply by @mpvl in https://github.com/cuelang/cue/pull/316#issuecomment-612194419_

This should be fixed as of v0.1.0.  Sorry about the confusion! 

Paul has also fixed the documentation infra so that we can have different versions for tip. Now all that remains is porting the k8s tutorial to cuelang.org.
